src/compute-client/src/plan/mod.rs

```diff
@@ -1570,14 +1570,41 @@ This is not expected to cause incorrect results, but could indicate a performanc
 
     /// Convert the dataflow description into one that uses render plans.
     #[tracing::instrument(
-        target = "optimizer"
+        target = "optimizer",
         level = "debug",
         skip_all,
-        fields(path.segment = "mir_to_lir")
+        fields(path.segment ="finalize_dataflow")
     )]
     pub fn finalize_dataflow(
         desc: DataflowDescription<OptimizedMirRelationExpr>,
         enable_monotonic_oneshot_selects: bool,
+    ) -> Result<DataflowDescription<Self>, String> {
+        // First, we lower the dataflow description from MIR to LIR.
+        let mut dataflow = Plan::lower_dataflow(desc)?;
+
+        // Subsequently, we perform plan refinements for the dataflow.
+        Plan::refine_source_mfps(&mut dataflow);
+
+        if enable_monotonic_oneshot_selects {
+            Plan::refine_single_time_dataflow(&mut dataflow);
+        }
```

Comment:
Nit: `Plan::` could be `Self::` in these three instances, I think.

(And these three methods could be free-standing functions, rather than members of `Plan`, but I realize that that's also true for `finalize_dataflow` itself. No reason to change anything here in this PR.)

Reply:
Cool, changed to `Self::` now.